ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topk/TopkKudaf.java

```diff
@@ -26,35 +29,33 @@
 import java.util.List;
 import java.util.Map;
 
-import io.confluent.ksql.function.KsqlAggregateFunction;
-import io.confluent.ksql.parser.tree.Expression;
-import io.confluent.ksql.util.ArrayUtil;
-import io.confluent.ksql.util.KsqlException;
-
-
 public class TopkKudaf<T> extends KsqlAggregateFunction<T, T[]> {
-
-  private Integer tkVal;
-  private T[] tempTopkArray;
-  private Class<T> ttClass;
+  private final int topKSize;
+  private final T[] tempTopKArray;
+  private final Class<T> clazz;
+  private final Schema initialValue;
+  private final List<Schema> arguments;
 
   TopkKudaf(int argIndexInValue,
-            Integer tkVal,
-            Class<T> ttClass) {
+            int topKSize,
+            Schema initialValue,
+            List<Schema> arguments,
+            Class<T> clazz) {
     super(argIndexInValue,
-          (T[]) Array.newInstance(ttClass,tkVal),
-          SchemaBuilder.array(Schema.FLOAT64_SCHEMA).build(),
```

Comment:
this line caused `ArrayStoreException` as Ksql engine implicitly converted `currentAggVal` variable to Double.